### PR TITLE
Unify ArrowTensorType tables and Tensor blocks

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

This PR removes the special-case Tensor block format from Datasets, in favor of representing these as single-column Arrow tables.

Before:
```
ray.data.range_tensor(10000)
# -> Dataset(num_blocks=200, num_rows=10000,
#            schema=<Tensor: shape=(None, 3, 5), dtype=int64>)
```

After:
```
ray.data.range_tensor(10000)
# -> Dataset(num_blocks=200, num_rows=10000,
#            schema={value: <ArrowTensorType: shape=(3, 5), dtype=int64>})

```

Read/write numpy operations have been changed to support Arrow tables with tensor columns.

This both removes a lot of special-case Tensor handling and cleans up the user story for tensor support. In the future, we can add more methods to make working with Tensors simpler on top of this common path.

### CODE REVIEW

**Comment by scv119** [comment]
looks like a great simplification! I'd defer this to @clarkzinzow to review and accept.

File: doc/source/data/dataset-tensor-support.rst [hunk 1]
```diff
@@ -3,66 +3,34 @@
 Dataset Tensor Support
 ======================
 
-Tensor-typed values
--------------------
+Tables with tensor columns
+--------------------------
+
+Datasets supports tables with fixed-shape tensor columns, where each element in the column is a tensor (n-dimensional array) with the same shape. As an example, this allows you to use both Pandas and Ray Datasets to read, write, and manipulate a table with a column of e.g. images (2D arrays), with all conversions between Pandas, Arrow, and Parquet, and all application of aggregations/operations to the underlying image ndarrays, being taken care of by Ray Datasets.
```
Comment (scv119) [inline doc/source/data/dataset-tensor-support.rst:9]: nit: the second sentence is a bit too long to understand.

Comment (ericl) [reply doc/source/data/dataset-tensor-support.rst]: Tried to optimize

File: python/ray/data/impl/arrow_block.py [hunk 2]
```diff
@@ -188,8 +185,21 @@ def schema(self) -> "pyarrow.lib.Schema":
...
+            raise ValueError(
+                "`column` must be specified when calling .to_numpy() "
+                "on Arrow blocks.")
+        if column not in self._table.column_names:
+            raise ValueError(
+                "Cannot find column {}, available columns: {}".format(
+                    column, self._table.column_names))
+        array = self._table[column]
+        if array.num_chunks > 1:
+            # TODO(ekl) combine fails since we can't concat ArrowTensorType?
```
Comment (clarkzinzow) [inline python/ray/data/impl/arrow_block.py:199]: Hmm should be easy to add, we can look at supporting this.

**Review by clarkzinzow — APPROVED** [review]
This looks great! I'm glad that it turned out to be a pretty simple port.

File: python/ray/data/impl/arrow_block.py (hunk 2 above)
Comment (ericl) [reply python/ray/data/impl/arrow_block.py:199]: Ok, do you have any pointers? I can give it a try

**Pull request merged**